pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularAgeValidationUpdate.java

```diff
@@ -0,0 +1,88 @@
+import org.broadinstitute.ddp.studybuilder.ActivityBuilder;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.sqlobject.SqlObject;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+import java.nio.file.Path;
+import java.util.List;
+
+/**
+ * Task to replace age validations to singular activities
+ */
+@Slf4j
+@NoArgsConstructor
+public class SingularAgeValidationUpdate implements CustomTask {
+    protected Path cfgPath;
+    protected Config cfg;
+    protected Config varsCfg;
+
+    @Override
+    public void init(final Path cfgPath, final Config studyCfg, final Config varsCfg) {
+        if (!studyCfg.getString("study.guid").equals("singular")) {
+            throw new DDPException("This task is only for the singular study!");
+        }
+
+        this.cfgPath = cfgPath;
+        this.cfg = studyCfg;
+        this.varsCfg = varsCfg;
+    }
+
+    @Override
+    public void run(final Handle handle) {
+        log.info("TASK::{}", SingularAgeValidationUpdate.class.getSimpleName());
+
+        final var study = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(cfg.getString("study.guid"));
+        final var user = handle.attach(JdbiUser.class).findByUserGuid(cfg.getString("adminUser.guid"));
+
+        final var builder = new ActivityBuilder(cfgPath.getParent(), cfg, varsCfg, study, user.getUserId());
+        updateValidations(handle, builder, "CONSENT_DEPENDENT", "patches/ddp-8554-dependent-validations.conf");
+        updateValidations(handle, builder, "CONSENT_PARENTAL", "patches/ddp-8554-parental-validations.conf");
+        updateValidations(handle, builder, "CONSENT_SELF", "patches/ddp-8554-self-validation.conf");
+    }
+
+    private void updateValidations(final Handle handle, final ActivityBuilder builder, final String activityCode, final String patchFile) {
+        final var file = cfgPath.getParent().resolve(patchFile).toFile();
+        if (!file.exists()) {
+            throw new DDPException("Data file is missing: " + patchFile);
+        }
+
+        updateValidations(handle,
+                builder,
+                handle.attach(JdbiActivity.class)
+                        .findActivityByStudyGuidAndCode(cfg.getString("study.guid"), activityCode)
+                        .orElseThrow(() -> new DDPException("Activity " + activityCode + " doesn't exist")),
+                ConfigFactory.parseFile(file).resolveWith(varsCfg));
+    }
+
+    private void updateValidations(final Handle handle, final ActivityBuilder builder, final ActivityDto activity, final Config config) {
+        final var activityVersion = handle.attach(JdbiActivityVersion.class)
+                .getActiveVersion(activity.getActivityId())
+                .orElseThrow(() -> new DDPException("Can't fetch latest activity version for " + activity.getActivityCode()));
+
+        handle.attach(SqlHelper.class).deleteValidations(activity.getActivityId());
```

Comment:
We probably don't have validations for the 3 activities we are updating but wondering why we have delete ? 
Deleting all activity_validations of the activity without any check !  
It works for rerunning the patch but is that the intent ?

Reply:
Since we don't have any other validations - yes. Besides if we figure out any other bugs later it will help us to replace validations easily. Anyway, as far as I understand the idea, this patch is going to be applied just once.